lib/ansible/plugins/action/network.py

```diff
@@ -110,6 +110,16 @@ def _handle_backup_option(self, result, task_vars):
         if copy_result.get('changed', False):
             result['changed'] = copy_result['changed']
 
+        if not backup_options or not backup_options.get('filename'):
```

Comment:
This condition can be simplified to be more readable
```
if backup_options and backup_options.get('filename'):
    result['date'] = time.strftime('%Y-%m-%d', time.gmtime(os.stat(result['backup_path']).st_ctime))
    result['date'] = time.strftime('%Y-%m-%d', time.gmtime(os.stat(result['backup_path']).st_ctime))
else:
    result['date'] = tstamp.split('@')[0]
    result['time'] = tstamp.split('@')[1]
    result['shortname'] = result['backup_path'][::-1].split('.', 1)[1][::-1]
    result['filename'] = result['backup_path'].split('/')[-1]  
```

Reply:
@ganeshrn Fixed.